boa_engine/src/vm/mod.rs

```diff
@@ -98,8 +102,19 @@ impl Vm {
     }
 }
 
+enum ShouldExit {
+    True,
+    False,
+    Yield,
+}
+
+pub(crate) enum ReturnType {
+    Normal,
+    Yield,
+}
+
```

Comment:
Are these two related to [completion records](https://github.com/boa-dev/boa/discussions/1541)? We should in any case add some documentation, and add `#[derive(Debug, Clone, Copy)]` to them.

Reply:
They are related, but not implemented like in the spec. As far as I understand it, the spec defines every return value as a completion record. I think we should re-evaluate if it makes sense / if we need to implement it that way.